components/brave_rewards/browser/publisher_info_database.cc

```diff
@@ -241,22 +241,50 @@ bool PublisherInfoDatabase::InsertOrUpdatePublisherInfo(
     return false;
   }
 
+  sql::Transaction transaction(&GetDB());
+  if (!transaction.Begin()) {
+    return false;
+  }
+
   sql::Statement publisher_info_statement(
       GetDB().GetCachedStatement(SQL_FROM_HERE,
                                  "INSERT OR REPLACE INTO publisher_info "
                                  "(publisher_id, verified, excluded, "
                                  "name, url, provider, favIcon) "
-                                 "VALUES (?, ?, ?, ?, ?, ?, ?)"));
+                                 "VALUES (?, ?, ?, ?, ?, ?, "
+                                 "(SELECT IFNULL( "
+                                 "(SELECT favicon FROM publisher_info "
+                                 "WHERE publisher_id = ?), \"\"))"
+                                 ")"));
 
   publisher_info_statement.BindString(0, info.id);
   publisher_info_statement.BindBool(1, info.verified);
   publisher_info_statement.BindInt(2, static_cast<int>(info.excluded));
   publisher_info_statement.BindString(3, info.name);
   publisher_info_statement.BindString(4, info.url);
   publisher_info_statement.BindString(5, info.provider);
-  publisher_info_statement.BindString(6, info.favicon_url);
+  publisher_info_statement.BindString(6, info.id);
+
+  publisher_info_statement.Run();
+
+  std::string favicon = info.favicon_url;
+  if (!favicon.empty()) {
+    sql::Statement favicon_statement(
+      GetDB().GetCachedStatement(SQL_FROM_HERE,
+                                 "UPDATE publisher_info SET favIcon = ? "
+                                 "WHERE publisher_id = ?"));
+
+    if (favicon == ledger::clear_favicon) {
+      favicon = "";
```

Comment:
Should we be doing something like `favicon.clear()` here? Same for other assignments to empty string

Reply:
not sure, I am used of just assigning stuff 😃 I can change it, don't have strong preference